x-pack/test/epm_api_integration/apis/ilm.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../api_integration/ftr_provider_context';
+import { getTemplate } from '../../../legacy/plugins/integrations_manager/server/lib/template/template';
+import { getPolicy } from '../../../legacy/plugins/integrations_manager/server/lib/ilm/ilm';
+
+export default function({ getService }: FtrProviderContext) {
+  describe('ilm', () => {
+
+    it('lists all packages from the registry', async () => {
+      const indexPattern = 'foo';
+      const templateName = 'bar';
+      const es = getService('es');
+
+      const body = getTemplate(indexPattern);
+
+      const response = await es.indices.putTemplate({
+        name: templateName,
+        body,
+      });
+      // Checks if template loading worked as expected
+      expect(response).to.eql({ acknowledged: true });
+
+      const indexTemplate = await es.indices.getTemplate({ name: templateName });
+      // Checks if the content of the template that was loaded is as expected
+      expect(indexTemplate[templateName].index_patterns).to.eql([indexPattern]);
+    });
+
+    it('setup policy', async () => {
+      const policyName = 'foo';
+      const es = getService('es');
+      const policy = getPolicy();
+
+      const response = await es.ilm.putLifecycle({
```

Comment:
@skh Based on the API I tried to use this method: https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#_ilm_putlifecycle But it tells me it doesn't exist yet. Which Elasticsearch client is used by Kibana and which version?

Reply:
You can check which dependencies / versions are used in the toplevel `package.json` file. You'll find two versions of `elasticsearch` there though, `16.5.0` in dependencies and `7.4.0` in dev dependencies. I'm right now trying to find out what that exactly means.

> But it tells me it doesn't exist yet.

What exactly tells you that what exactly doesn't exist? How is this related to the version?

Reply:
Update: kibana uses elasticsearch-js `16.5.0` built from https://github.com/elastic/elasticsearch-js-legacy. Switching to the new client is tracked in https://github.com/elastic/kibana/issues/35508 .

Reply:
I checked in my local setup `node_modules/elasticsearch` and in there is not what I really expected when looking at https://github.com/elastic/elasticsearch-js Now looking at what you mentioned with the dev dependencies, what I would need is 7.4.0, I assume 16.5.0 comes from an old version? https://github.com/elastic/elasticsearch-js/releases There is 16.x release on the top.

Reply:
@skh Thank you, exactly what I was looking for.